cypress/e2e/GUI/gui.cy.js

```diff
@@ -0,0 +1,141 @@
+        cy.contains('Hide address').click()
+        cy.contains('Show address').should('be.visible')
+    })
+
+    it('Retorna a lista de clientes ao clicar no botao "Voltar"', () => {
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+        cy.contains('button', 'Back').click()
+        cy.contains('Below is our customer list.').should('be.visible')
+    })
+
+    it('Abre e fecha o messenger', () => {
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('[aria-label="Close messenger"]').should('be.visible')
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('[aria-label="Open messenger"]').should('be.visible')
+    })
+
+    it('Garante que todos os campos do messenger sao obrigatorios e que o primeiro esta focado', () => {
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('[class^="Messenger_form"]')
+            .find('input[type="text"]')
+            .should('be.focused')
+            .and('have.attr', 'required')
+        cy.get('[class^="Messenger_form"]')
+            .find('input[type="email"]')
+            .should('have.attr', 'required')
+        cy.get('[class^="Messenger_form"]')
+            .find('textarea')
+            .should('have.attr', 'required')
+    })
+
+    it('Verifica se e possivel enviar uma mensagem pelo messenger com e-mail invalido', () => {
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('[aria-label="Close messenger"]').should('be.visible')
+        cy.get('input[id="messenger-name"]').type('Joe')
+        cy.get('input[id="email"]').type('testexample.com')
+        cy.get('textarea[id="message"]').type('Test')
+        cy.get('input[type="email"]').should('be.visible')
+    })
+
+    it('Mostra e oculta uma mensagem de sucesso ao enviar o formulario do messenger com sucesso', () => {
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.clock()
+        cy.get('input[id="messenger-name"]').type('Joe')
+        cy.get('input[id="email"]').type('test@example.com')
+        cy.get('textarea[id="message"]').type('Test')
+        cy.get('button[type="submit"]').click()
+        cy.contains('Your message has been sent').should('be.visible')
+        cy.tick(3000)
+        cy.contains('div.Messenger_success_nqnL1').should('not.exist')
+    })
+
+    it('Limpa todos os campos do formulario do messenger ao preenche-los, fechar o messenger e abri-lo novamente', () => {
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('input[id="messenger-name"]').type('Joe')
+        cy.get('input[id="email"]').type('test@example.com')
+        cy.get('textarea[id="message"]').type('Test')
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('[class^="Messenger_openCloseButton"]').click()
+        cy.get('input[id="messenger-name"]').should('be.empty')
```

Comment:
A verificação `.should('be.empty')` não é confiável.

Execute este teste comentando as linhas 135 e 136 e verás que esta e a verificação abaixo vão passar, e só a do campo de area de texto (`textarea`) irá falhar.

Portanto, use `.should('have.value', '')`.

Tal verificação analisa especificamente o valor de cada campo, tornando o teste mais confiável.

Me lembre de falar sobre resultados falsos positivos no próximo encontro.